### PR TITLE
fix(curriculum) - Correct HTML Forms registration step 17

### LINKED ISSUE #49261: Learn HTML Forms Registration Form - Step 17
### Describe the Issue

Instructions ask for nested \<input\> elements within each \<label\>, but do not allow multi-line responses. When the response is submitted with multiple nested line, I get the error:

Test
Sorry, your code does not pass. You're getting there.

Hint
You should add the first input after the label text Enter Your Last Name:, and include a space after the colon.

### Affected Page

https://www.freecodecamp.org/learn/2022/responsive-web-design/learn-html-forms-by-building-a-registration-form/step-17

### Your code

Fails:
```
      <fieldset>
        <label>Enter Your First Name: 
          <input />
        </label>
        <label>Enter Your Last Name: 
          <input />
        </label>
        <label>Enter Your Email: 
          <input />
        </label>
        <label>Create a New Password: 
          <input />
        </label>
      </fieldset>
```

Passes:
```
      <fieldset>
        <label>Enter Your First Name: <input /></label>
        <label>Enter Your Last Name: <input /></label>
        <label>Enter Your Email: <input /></label>
        <label>Create a New Password: <input /></label>
      </fieldset>
```

### Expected behavior

Submitting code should allow for the case of multi-line nesting. (The first segment of failing code above should pass.)

### Screenshots

_No response_

### System

- Device: MacBook Pro
- OS: OS X 10.15
- Browser: Brave
- Version: 1.47.171 Chromium: 109.0.5414.87 (Official Build) (x86_64)

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by a2937**
Checklist:

- [X] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [X] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [X] I have tested these changes either locally on my machine, or GitPod.

Closes #49261

I replaced the standard equals method with a Regex query which should be a lot easier to deal with.

### CODE REVIEW

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/60f805f813eaf2049bc2ceea.md [hunk 1]
```diff
@@ -26,25 +26,33 @@ assert.equal(document.querySelectorAll('label input')?.length, 4);
 You should add the first `input` after the `label` text `Enter Your First Name:`, and include a space after the colon.
 
 ```js
-assert.equal(document.querySelectorAll('label')?.[0]?.innerHTML.trim(), 'Enter Your First Name: <input>');
+const query = /Enter Your First Name:\s*<input>/
```
Comment (jeremylt) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/60f805f813eaf2049bc2ceea.md:29]: Will this allow for people writing their input tags as `<input/>`?

I thought we generally wanted to use the `querySelectorAll` functionality?

Comment (a2937) [reply curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/60f805f813eaf2049bc2ceea.md:29]: For the first question, yes. 

As for the second question, document.querySelectorAll is being used down below. What am I doing is performing a match against the innerHtml to the query I've made in order to handle any additional spacing problems.

Comment (jeremylt) [reply curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/60f805f813eaf2049bc2ceea.md:29]: I deleted the second question - I understood what I read a bit better after I posted the question

Comment (a2937) [reply curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/60f805f813eaf2049bc2ceea.md:29]: > Will this allow for people writing their input tags as `<input/>`?

Here's a picture btw. 

Screenshot

**Comment by a2937** [comment]
Thank you for pointing this out. It now requires there to be at least one space after the colon and that there shouldn't be any extraneous characters in the html.

**Review by moT01 — APPROVED** [review]
LGTM @a2937 🎉 Thanks for contributing to freeCodeCamp 🎉

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/60f805f813eaf2049bc2ceea.md [hunk 2]
```diff
@@ -26,25 +26,29 @@ assert.equal(document.querySelectorAll('label input')?.length, 4);
 You should add the first `input` after the `label` text `Enter Your First Name:`, and include a space after the colon.
 
 ```js
-assert.equal(document.querySelectorAll('label')?.[0]?.innerHTML.trim(), 'Enter Your First Name: <input>');
+const query = /^Enter Your First Name:\s+<input>/
+assert.equal(query.test(document.querySelectorAll('label')?.[0]?.innerHTML.trim()),true);
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/60f805f813eaf2049bc2ceea.md:30]:
```suggestion
assert.match(document.querySelectorAll('label')?.[0]?.innerHTML.trim(), query);
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/60f805f813eaf2049bc2ceea.md [hunk 3]
```diff
@@ -26,25 +26,29 @@ assert.equal(document.querySelectorAll('label input')?.length, 4);
...
 ```
 
 You should add the second `input` after the `label` text `Enter Your Last Name:`, and include a space after the colon.
 
 ```js
-assert.equal(document.querySelectorAll('label')?.[1]?.innerHTML.trim(), 'Enter Your Last Name: <input>');
+const query = /^Enter Your Last Name:\s+<input>/
+assert.equal(query.test(document.querySelectorAll('label')?.[1]?.innerHTML.trim()),true);
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/60f805f813eaf2049bc2ceea.md:37]:
```suggestion
assert.match(document.querySelectorAll('label')?.[1]?.innerHTML.trim(), query);
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/60f805f813eaf2049bc2ceea.md [hunk 4]
```diff
@@ -26,25 +26,29 @@ assert.equal(document.querySelectorAll('label input')?.length, 4);
...
 ```
 
 You should add the third `input` after the `label` text `Enter Your Email:`, and include a space after the colon.
 
 ```js
-assert.equal(document.querySelectorAll('label')?.[2]?.innerHTML.trim(), 'Enter Your Email: <input>');
+const query = /^Enter Your Email:\s+<input>/
+assert.equal(query.test(document.querySelectorAll('label')?.[2]?.innerHTML.trim()),true);
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/60f805f813eaf2049bc2ceea.md:44]:
```suggestion
assert.match(document.querySelectorAll('label')?.[2]?.innerHTML.trim(), query);
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/60f805f813eaf2049bc2ceea.md [hunk 5]
```diff
@@ -26,25 +26,29 @@ assert.equal(document.querySelectorAll('label input')?.length, 4);
...
 ```
 
 You should add the fourth `input` after the `label` text `Create a New Password:`, and include a space after the colon.
 
 ```js
-assert.equal(document.querySelectorAll('label')?.[3]?.innerHTML.trim(), 'Create a New Password: <input>');
+const query = /^Create a New Password:\s+<input>/
+assert.equal(query.test(document.querySelectorAll('label')?.[3]?.innerHTML.trim()),true);
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/60f805f813eaf2049bc2ceea.md:51]:
```suggestion
assert.match(document.querySelectorAll('label')?.[3]?.innerHTML.trim(), query);
```

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
Using the `match` assertion is ideal here.

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/60f805f813eaf2049bc2ceea.md (hunk 2 above)
Comment (a2937) [reply curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/60f805f813eaf2049bc2ceea.md:30]: Woah. I didn't even know that method was a thing. All I relatively know was the Chai assertion library.

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/60f805f813eaf2049bc2ceea.md (hunk 3 above)
Comment (a2937) [reply curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/60f805f813eaf2049bc2ceea.md]: Done.

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/60f805f813eaf2049bc2ceea.md (hunk 5 above)
Comment (a2937) [reply curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/60f805f813eaf2049bc2ceea.md:51]: Done.

**Review by bbsmooth — APPROVED** [review]
I tried really hard to break this, but alas, it was able to handle everything I could throw at it.

LGTM.

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/60f805f813eaf2049bc2ceea.md (hunk 5 above)
Comment (naomi-lgbt) [reply curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/60f805f813eaf2049bc2ceea.md:51]: This one appears to still be assert.equal

Comment (a2937) [reply curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/60f805f813eaf2049bc2ceea.md:51]: Thanks Naomi. I missed that one by accident.

**Review by naomi-lgbt — APPROVED** [review]
image

**Pull request merged**